packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,155 @@
+      const whoLeft = previousParticipants.filter((p) => !currentParticipants.includes(p));
+      if (whoJoined.length > whoLeft.length) {
+        resetAnnoucement(createAnnouncmentString('joined', whoJoined, strings));
+      }
+      if (whoLeft.length > whoJoined.length) {
+        resetAnnoucement(createAnnouncmentString('left', whoLeft, strings));
+      }
+      // Update cached value at the end.
+      setPreviousParticipants(currentParticipants);
+    },
+    // previousParticipants caches the value of `currenParticipants`. We _don't_ want this
+    // hook to run for when `previousParticipants` is updated.
+    // If we did, the second run would always clear out the value of `whoJoined` etc.
+    [currentParticipants, previousParticipants, strings]
+  );
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+export const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  /**
+   * Filter participants into two arrays to put all the unnamed participants at the back of the
+   * names being announced.
+   */
+  const unnamedParticipants = participants.filter((p) => p.displayName === undefined);
+  const namedParicipants = participants.filter((p) => p.displayName);
+  const sortedParticipants = namedParicipants.concat(unnamedParticipants);
+
+  /**
+   * if there are only unnamed participants present in the array announce a special unnamed participants
+   * only string.
+   */
+  if (sortedParticipants.filter((p) => p.displayName).length === 0 && sortedParticipants.length > 1) {
+    return _formatString(
+      direction === 'joined' ? strings.manyUnnamedParticipantsJoined : strings.manyUnnamedParticipantsLeft,
+      {
+        numOfParticipants: (sortedParticipants.length - 1).toString()
+      }
+    );
+  }
+  const participantNames = sortedParticipants.map((p) => p.displayName ?? strings.unnamedParticipantString);
+
+  switch (sortedParticipants.length) {
+    case 1:
+      return _formatString(
+        direction === 'joined' ? strings.participantJoinedNoticeString : strings.participantLeftNoticeString,
+        { displayName: participantNames[0] }
+      );
+    case 2:
+      return _formatString(
+        direction === 'joined' ? strings.twoParticipantJoinedNoticeString : strings.twoParticipantLeftNoticeString,
+        {
+          displayName1: participantNames[0],
+          displayName2: participantNames[1]
+        }
+      );
+    case 3:
+      return _formatString(
+        direction === 'joined' ? strings.threeParticipantJoinedNoticeString : strings.threeParticipantLeftNoticeString,
+        {
+          displayName1: participantNames[0],
+          displayName2: participantNames[1],
+          displayName3: participantNames[2]
+        }
+      );
+  }
```

Comment:
nit: add a `case 0` to this to return blank or throw an error as the code that follows this assumes we have at least 3 participants and this function never guaranteed that. Alternatively at the start of this function have a `if (participants.length === 0) return;`